dmutils/filters.py

```diff
@@ -24,19 +25,23 @@ def format_links(text):
                                 \.                        # match dot in host name
                                 (?:[^\s/?\.#<>"']+\.?)+   # match rest of domain
                                 (?:/[^\s<>"']*)?          # rest of url
-                                [^\s<>"'\.]               # no dot at end
+                                [^\s<>,"'\.]              # no dot at end
                                 )""", re.X)
-    matches = url_match.findall(text)
-    link = '{0}<a href={1} rel="external">{1}</a>'
-    if matches:
-        textArray = url_match.split(text)
-        formattedText = ""
-        for text in textArray:
-            if text in matches:
-                formattedText = link.format(formattedText, Markup.escape(text))
+    matched_urls = url_match.findall(text)
+    if matched_urls:
+        link = '<a href="{0}" rel="external">{1}</a>'
+        formatted_text = ""
+        text_array = url_match.split(text)
+        formatted_text_array = []
+        for partial_text in text_array:
+            if partial_text in matched_urls:
+                url = "https://{}".format(partial_text) if partial_text.startswith('www') else partial_text
```

Comment:
I don't think we can do https here, as it won't work for any sites that don't have SSL/TLS.  Doing http is possibly less secure, but probably slightly more reliable.